src/github_runner_manager/reactive/consumer.py

```diff
@@ -56,11 +96,75 @@ def consume(mongodb_uri: str, queue_name: str) -> None:
                     msg.reject(requeue=True)
                     raise JobError(f"Invalid job details: {msg.payload}") from exc
                 logger.info(
-                    "Received job with labels %s and run_url %s",
+                    "Received job with labels %s and job_url %s",
                     job_details.labels,
-                    job_details.run_url,
+                    job_details.job_url,
                 )
-                msg.ack()
+                _spawn_runner(
+                    runner_manager=runner_manager,
+                    job_url=job_details.job_url,
+                    msg=msg,
+                    github_client=github_client,
+                )
+
+
+def _spawn_runner(
+    runner_manager: RunnerManager, job_url: HttpUrl, msg: Message, github_client: GithubClient
+) -> None:
+    """Spawn a runner.
+
+    A runner is only spawned if the job has not yet been picked up by a runner.
+    After spawning a runner, it is checked if the job has been picked up.
+
+    If the job has been picked up, the message is acknowledged.
+    If the job has not been picked up after 5 minutes, the message is rejected and requeued.
+
+    Args:
+        runner_manager: The runner manager to use.
+        job_url: The URL of the job.
+        msg: The message to acknowledge or reject.
+        github_client: The GitHub client to use to check the job status.
+    """
+    if _check_job_been_picked_up(job_url=job_url, github_client=github_client):
+        msg.ack()
+        return
+    runner_manager.create_runners(1)
+    for _ in range(10):
+        if _check_job_been_picked_up(job_url=job_url, github_client=github_client):
+            msg.ack()
+            break
+        sleep(30)
+    else:
+        msg.reject(requeue=True)
+
+
+def _check_job_been_picked_up(job_url: HttpUrl, github_client: GithubClient) -> bool:
+    """Check if the job has already been picked up.
+
+    Args:
+        job_url: The URL of the job.
+        github_client: The GitHub client to use to check the job status.
+
+    Returns:
+        True if the job has been picked up, False otherwise.
+    """
+    # job_url has the format:
+    # "https://api.github.com/repos/cbartz/gh-runner-test/actions/jobs/22428484402"
+    path = job_url.path
+    # we know that path is not empty as it is validated by the JobDetails model
+    job_url_path_parts = path.split("/")  # type: ignore
+    job_id = job_url_path_parts[-1]
+    owner = job_url_path_parts[2]
+    repo = job_url_path_parts[3]
+    logging.debug(
+        "Parsed job_id: %s, owner: %s, repo: %s from job_url path %s", job_id, owner, repo, path
+    )
```

Comment:
We plan to pass the job_id and owner/repo in the queue instead of the job_url (cc @gregory-schiano ), which would remove the need to split the url into components here.